packages/webarchitect/src/validateDirectoryRule.ts

```diff
@@ -0,0 +1,160 @@
+import { stat } from "node:fs/promises";
+import { join } from "node:path";
+import listDirectory from "./listDirectory.js";
+import type { DirectoryRule, ValidationResult } from "./types.js";
+import validateFileRule from "./validateFileRule.js";
+
+/**
+ * Validates a directory against a directory rule specification.
+ * Checks that the directory exists and validates its contents recursively.
+ * In strict mode, also ensures no unexpected files or directories are present.
+ *
+ * @param projectPath - Base path where the directory should exist
+ * @param dirRule - Directory rule specification including name and expected contents
+ * @param ruleName - Name of the rule for error reporting (e.g., "src-structure")
+ * @returns Promise that resolves to an array of validation results
+ *
+ * @example
+ * ```typescript
+ * const rule = {
+ *   name: "src",
+ *   contains: {
+ *     files: [{ name: "index.ts", contains: {...} }],
+ *     directories: [{ name: "components", contains: {...} }]
+ *   },
+ *   strict: true
+ * };
+ * const results = await validateDirectoryRule("/project", rule, "source-structure");
+ * ```
+ */
+export default async function validateDirectoryRule(
+  projectPath: string,
+  dirRule: DirectoryRule,
+  ruleName: string, // Add rule name parameter
+): Promise<ValidationResult[]> {
+  const dirPath = join(projectPath, dirRule.name);
+  const results: ValidationResult[] = [];
+
+  // Check if directory exists
+  try {
+    const stats = await stat(dirPath);
+    if (!stats.isDirectory()) {
+      return [
+        {
+          rule: ruleName, // Use the rule name from schema
+          passed: false,
+          message: `Expected directory but found file: ${dirPath}`,
+          context: {
+            type: "directory" as const,
+            target: dirPath,
+            description: `Validates directory structure for ${dirRule.name}`,
+            actualValue: "[File found instead of directory]",
+          },
+        },
```

Comment:
same comment as in validateFileRule - these are constructed repetitively with the same values, would recommend DRYing